packages/epo-widget-lib/src/lib/utils.ts

```diff
@@ -0,0 +1,18 @@
+/** clamps the index to the bounds of an array
+ *  if the index is higher than the final index
+ *  it will clamp to 0, if it is lower it will clamp
+ *  to the final index.
+ */
+export const getClampedArrayIndex = (
```

Comment:
replaced the variable mutation going on in the main component with this method that checks if the index is higher/lower than the array range and returns a new index accordingly